storage/storage_helpers.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"database/sql"
+	"fmt"
+	time "time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/google/trillian"
+	"github.com/google/trillian/crypto/keyspb"
+	spb "github.com/google/trillian/crypto/sigpb"
+)
+
+// ToMillisSinceEpoch converts a timestamp into milliseconds since epoch
+func ToMillisSinceEpoch(t time.Time) int64 {
+	return t.UnixNano() / 1000000
+}
+
+// FromMillisSinceEpoch converts an UNIX typestamp to a time struct
+func FromMillisSinceEpoch(ts int64) time.Time {
+	secs := int64(ts / 1000)
+	msecs := int64(ts % 1000)
+	return time.Unix(secs, msecs*1000000)
+}
+
+// SetNullStringIfValid assigns src to dest if src is Valid.
+func SetNullStringIfValid(src sql.NullString, dest *string) {
+	if src.Valid {
+		*dest = src.String
+	}
+}
+
+// ValidateStorageSettings checks for storage settings and returns an error if they exist
+func ValidateStorageSettings(tree *trillian.Tree) error {
```

Comment:
It doesn't seem like this should be a general helper, since storage_settings is supposed to be specific to individual storage implementations.